scipy/integrate/tests/test_quadrature.py

```diff
@@ -730,6 +634,18 @@ def test_simpson_exceptions(self, message, kwarg_update):
         with pytest.raises(ValueError, match=message):
             cumulative_simpson(**dict(kwargs0, **kwarg_update))
 
+    def test_special_cases(self):
+        # Test special cases not checked elsewhere
+        rng = np.random.default_rng(82456839535679456794)
+        y = rng.random(size=10)
+        res = cumulative_simpson(y, dx=0)
+        assert_equal(res, 0)
+
+        # Should add tests of:
+        # - all elements of `x` identical
+        # - y of size 1 and 2
+        # These should work as they do for `simpson`
```

Comment:
I put the special case of `dx=0` in a separate test so that it's more visible as such. This brings to mind that there are some other special cases that would ideally return the correct result rather than being rejected by input validation. We don't have to take care of them right now, but it would be a good follow-up.